server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodes.java

```diff
@@ -99,7 +99,14 @@ public Iterator<DiscoveryNode> iterator() {
         return nodes.values().iterator();
     }
 
-    @Override
+    public Stream<DiscoveryNode> stream() {
+        return nodes.values().stream();
+    }
+
+    public Collection<DiscoveryNode> getAllNodes() {
+        return Collections.unmodifiableCollection(nodes.values());
```

Comment:
I don't think you need to wrap unmodifiable here, the nodes are always immutable because the builder for this class does a `Map.copyOf(`.